file.cc

```diff
@@ -51,179 +33,96 @@ bool tryPossibleZipName( std::string const & name, std::string & copyTo )
     return false;
 }
 
-void loadFromFile( std::string const & n, std::vector< char > & data )
+void loadFromFile( std::string const & filename, std::vector< char > & data )
 {
-  File::Class f( n, "rb" );
-
-  f.seekEnd();
-
-  data.resize( f.tell() );
-
-  f.rewind();
-
-  f.read( &data.front(), data.size() );
-}
-
-bool exists( char const * filename ) noexcept
-{
-#ifdef __WIN32
-#if defined(__WIN64) || defined(_MSC_VER)
-  struct _stat64 buf;
-#else
-  struct _stat buf;
-#endif
-  wchar_t wname[16384];
-  MultiByteToWideChar( CP_UTF8, 0, filename, -1, wname, 16384 );
-#if defined(__WIN64) || defined(_MSC_VER)
-  return _wstat64( wname, &buf ) == 0;
-#else
-  return _wstat( wname, &buf ) == 0;
-#endif
-#else
-  struct stat buf;
-
-  // EOVERFLOW rationale: if the file is too large, it still does exist
-  return stat( filename, &buf ) == 0 || errno == EOVERFLOW;
-#endif
+  File::Class f( filename, "rb" );
+  QByteArray byteArray{ f.readall() };
+  data.reserve( byteArray.size() );
+  data = std::vector< char >( byteArray.cbegin(), byteArray.cend() );
```

Comment:
https://doc.qt.io/qt-6/qbytearray.html#constData

does this help to reduce the copy

Reply:
I asked a SO question: 

https://stackoverflow.com/questions/76019968/optimal-way-to-convert-qbytearray-to-stdvectorchar

The one in the code should be the best.

---

The `loadFromFile()` seems exclusively used to load resources like images & audio. I think we are using Qimage to process them. Maybe this method should be removed, and return Qimage directly. So that the conversion/copying will be completely gone. Consider in another PR :)

Obtain a generic type then process in each dict file seems not elegant.

Reply:
I got it ,if it used to hold images ,  constData should not be used.

btw,
![image](https://user-images.githubusercontent.com/105986/232182822-43fc4b66-0363-44c0-8a70-3a3f221689c2.png)

Reply:
There are 5 use of this function, and they are all related to load image files.

I think we should just replace this function with `File::loadImage() -> QImage` to have absolutely zero overhead.

In another PR?

Reply:
another PR is fine .